editor/editor_node.cpp

```diff
@@ -5100,7 +5136,57 @@ void EditorNode::_dock_tab_changed(int p_tab) {
 	}
 }
 
+bool EditorNode::_restore_floating_dock(const Dictionary &p_dock_dump, Control *p_dock, int p_slot_index) {
+	// Restore the detached window.
+	Rect2i window_rect = p_dock_dump.get("window_rect", Rect2i());
+	int screen = p_dock_dump.get("window_screen", -1);
+	Rect2i screen_rect = p_dock_dump.get("window_screen_rect", Rect2i());
+
+	if (screen < 0 || screen >= DisplayServer::get_singleton()->get_screen_count()) {
+		// Fallback to the main window screen if the saved screen is not available.
+		screen = get_viewport()->get_window_id();
+	}
+
+	Rect2i real_screen_rect = DisplayServer::get_singleton()->screen_get_usable_rect(screen);
+
+	if (screen_rect == Rect2i()) {
+		// Fallback to the target screen rect.
+		screen_rect = real_screen_rect;
+	}
+
+	if (window_rect == Rect2i()) {
+		// Fallback to a standard rect.
+		window_rect = Rect2i(screen_rect.position + screen_rect.size / 4, screen_rect.size / 2);
+	}
+
+	// Adjust the window rect size in case the resolution changes.
+	Vector2 screen_ratio = Vector2(real_screen_rect.size) / Vector2(screen_rect.size);
+
+	WindowWrapper *wrapper = Object::cast_to<WindowWrapper>(p_dock);
+	if (!wrapper) {
+		_dock_make_float(p_dock, p_slot_index);
+		wrapper = Object::cast_to<WindowWrapper>(floating_docks[floating_docks.size() - 1]);
+	}
+
+	// The screen positioning may change, so remove the original screen position.
+	window_rect.position -= screen_rect.position;
+	window_rect = Rect2i(window_rect.position * screen_ratio, window_rect.size * screen_ratio);
+	window_rect.position += real_screen_rect.position;
+
+	// All good, restore the window.
+	wrapper->restore_window(window_rect, screen);
+
+	return true;
+}
+
 void EditorNode::_load_docks_from_config(Ref<ConfigFile> p_layout, const String &p_section) {
+	Dictionary floating_docks_dump;
+	if (p_layout->has_section_key(p_section, "dock_floating")) {
+		floating_docks_dump = p_layout->get_value(p_section, "dock_floating");
+	}
```

Comment:
Can be simplified to
```suggestion
	floating_docks_dump = p_layout->get_value(p_section, "dock_floating", Dictionary());
```